inst/htmlwidgets/plotly.js

```diff
@@ -50,7 +50,7 @@ HTMLWidgets.widget({
           var tr = x.data[pt.curveNumber];
           // add on additional trace info, if it exists
           attachKey = function(keyName) {
-            if (tr.hasOwnProperty(keyName)) {
+            if (tr.hasOwnProperty(keyName) && tr[keyName] !== null) {
               if (typeof pt.pointNumber === "number") {
                 obj[keyName] = tr[keyName][pt.pointNumber];
               } else {
```

Comment:
I was having trouble running https://gist.github.com/walkerke/da941dac84f730adcdde locally and discovered that it was because `key` is defined, but `null` (a result of recent changes to the R client). This provides a safer check before trying to subset trace info.

@timelyportfolio I'm a JS novice, so if you see anything in `inst/htmlwidgets/plotly.js` that could be improved, please let me know :)